sotodlib/site_pipeline/make_ml_map.py

```diff
@@ -0,0 +1,322 @@
+    parser.add_argument("--prefix", type=str)
+    parser.add_argument('-C', "--comps",   type=str, help="T,Q, and/or U")
+    parser.add_argument("-c", "--context", type=str,  help="Context containing TODs")
+    parser.add_argument("-n", "--ntod",    type=int)
+    parser.add_argument(      "--tods",    type=str)
+    parser.add_argument(      "--nset",    type=int)
+    parser.add_argument("-N", "--nmat",    type=str)
+    parser.add_argument(      "--max-dets",type=int)
+    parser.add_argument("-S", "--site",    type=str)
+    parser.add_argument("-v", "--verbose", action="count")
+    parser.add_argument("-q", "--quiet",   action="count")
+    parser.add_argument("-@", "--center-at", type=str)
+    parser.add_argument("-w", "--window",  type=float)
+    parser.add_argument("-i", "--inject",  type=str)
+    parser.add_argument(      "--nocal",   action="store_true", default=True)
+    parser.add_argument(      "--nmat-dir", type=str)
+    parser.add_argument(      "--nmat-mode", type=str, help="How to build the noise matrix. 'build': Always build from tod. 'cache': Use if available in nmat-dir, otherwise build and save. 'load': Load from nmat-dir, error if missing. 'save': Build from tod and save.")
+    parser.add_argument("-d", "--downsample", type=int, help="Downsample TOD by this factor")
+    parser.add_argument(      "--maxiter",    type=int)
+    parser.add_argument("-T", "--tiled"  ,    type=int)
+    parser.add_argument("-W", "--wafer"  ,   type=str, nargs='+', help="Detector wafer subset to map with")
+    return parser
+
+
+def _get_config(config_file):
+    return yaml.safe_load(open(config_file,'r'))
+
+
+
+def main(config_file=None, defaults=defaults, **args):
+    
+    cfg = defaults
+    if config_file is not None:
+        # Load in config file
+        cfg_from_file = _get_config(config_file)
+        
+        # Check which fields have been supplied in config.yaml
+        for field, file_val in cfg_from_file.items():
+            # If an optional arg is supplied, in the config file
+            # add it here
+            if field not in cfg.keys():
+                cfg[field] = cfg_from_file[field]
+            # If the value in the config file is different from the 
+            # default, update it
+            elif cfg[field] != file_val:
+                cfg[field] = file_val
+    else:
+        print("No config file provided, assuming default values") 
+
+    # Merge flags from config file and defaults with any passed through CLI
+    for key, item in args.items():
+        if item is not None:
+            cfg[key] = item
+    # Certain fields are required. Check if they are all supplied here
+    required_fields = ['query','freq','area','odir','prefix','context']
+    for req in required_fields:
+        if req not in cfg.keys():
+            raise KeyError("{} is a required argument. Please supply it in a config file or via the command line".format(req))
+
+    args = cfg
```

Comment:
To clarify the intended workflow here:
1. Load in the default values into `cfg`
2. Check for a config file. If one is found, load it in.
3. If the config file had key/values not found in the default dict, add them
4. If the config file had values that do not match the default dict's values, update `cfg` with those values
5. Update `cfg` with values supplied in the command line
6. Throw an error if one of the crucial required values is not supplied
7. Continue to mapmaking